### PR TITLE
update Substrate/Polkadot/Cumulus refs

### LINKED ISSUE #1458: Backport GRANDPA tests fixes
There are some changes in https://github.com/paritytech/polkadot/pull/5667 (more details in linked Substrate PR). We need to backport it before it reach Kusama <> Polkadot. Right now it is problematic, since we depend on XCM v3 branches

### LINKED ISSUE #1534: Update dependecies before going live on Rococo/Wococo
**Bridges repo:**
- [ ] update substrate `sv-locked-for-gav-xcm-v3-and-bridges` to actual `master`
- [ ] update polkadot `locked-for-gav-xcm-v3-and-bridges` to actual `gav-xcm-v3` or if merged to `master`
- [ ] fix bridges's master
- [ ] when bridges's master work, then rebase/merge `bko-bridge-ro-wo`

**Cumulus repo:**

- [ ] update bridges `git subtree` repo to actual `master` (temporary using `bko-bridge-ro-wo`)

### PR DESCRIPTION
**Pull request by svyatonik**
closes #1534
closes #1458 

TODOs:
- [x] fix compilation;
- [x] backport GRANDPA fixes;
- [ ] get rid of `*-2` branches in Substrate and Polkadot;
- [x] verify that it works.

cc @bkontur I'm updating to latest commit of Polkadot' `gav-xcm-v3`, Substrate commit that it references and `gav-xcm-v3` of Cumulus. Will it help you?

### CODE REVIEW

**Comment by svyatonik** [comment]
I'll do the third point (get rid of *-2 branches in Substrate and Polkadot) in a follow up PR, since review may take some time and force-pushing to those branches would break the build

File: bin/millau/runtime/src/lib.rs [hunk 1]
```diff
@@ -748,37 +753,72 @@ impl_runtime_apis! {
...
+			-> Result<(Vec<mmr::EncodableOpaqueLeaf>, mmr::BatchProof<MmrHash>), mmr::Error>
+		{
+			Mmr::generate_batch_proof(leaf_indices)
+				.map(|(leaves, proof)| (leaves.into_iter().map(|leaf| mmr::EncodableOpaqueLeaf::from_leaf(&leaf)).collect(), proof))
+		}
+
+		fn verify_batch_proof(leaves: Vec<mmr::EncodableOpaqueLeaf>, proof: mmr::BatchProof<MmrHash>)
+			-> Result<(), mmr::Error>
+		{
+			type Leaf = <
```
[the remark this answers is not in the document]
Comment (svyatonik) [reply bin/millau/runtime/src/lib.rs:804]: Yeah - I did that inititally and obviously the proper name is `MmrLeaf`, but it is already used by the `pallet_beefy_mmr` pallet, so imo let's leave as is

File: bin/millau/runtime/src/lib.rs [hunk 2]
```diff
@@ -748,37 +753,72 @@ impl_runtime_apis! {
...
+				vec![node],
+				pallet_mmr::primitives::Proof::into_batch_proof(proof),
+			)
 		}
 
 		fn mmr_root() -> Result<MmrHash, MmrError> {
 			Ok(Mmr::mmr_root())
 		}
+
+		fn generate_batch_proof(leaf_indices: Vec<pallet_mmr::primitives::LeafIndex>)
```
[the remark this answers is not in the document]
Comment (svyatonik) [reply bin/millau/runtime/src/lib.rs:794]: We are just trying to copy code from Substrate/Polkadot/Cumulus runtimes/nodes here. Ideally we should just copy original code, add our pallets and APIs and that's it. But actually we have already diverged a lot and updating runtimes/nodes isn't a trivial task anymore. Adding our local macro would make it even harder - we'll just diverge again from the original code. If you want' you may open an issue/PR in the Substrate repo with such macro

[the remark this answers is not in the document]
Comment (serban300) [reply bin/millau/runtime/src/lib.rs:794]: Ok, makes sense. I'll take a more detailed look in the Substrate repo and see if it makes sense to add a macro there.

File: bin/millau/runtime/src/lib.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (serban300) [reply bin/millau/runtime/src/lib.rs:804]: Maybe we can use the same approach as in [substrate](https://github.com/paritytech/substrate/blob/20037d573c0c2b319998541c6140df4c3d07eb56/bin/node/runtime/src/lib.rs#L1700) (declaring it inside `mod mmr`).

File: bin/millau/runtime/src/lib.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (serban300) [reply bin/millau/runtime/src/lib.rs:794]: Maybe we could implement some things in the bridges repo as well to make these upgrades easier. I'm thinking. I will do some experiments. But I have one more question. Where are we copying this `impl_runtime_apis!` block from ? Is it an identical copy to which we add our custom APIs at the end ? Or is it copied and adapted ?

[the remark this answers is not in the document]
Comment (svyatonik) [reply bin/millau/runtime/src/lib.rs:794]: Both Rialto+Millau have been forked from `substrate-node-template`, when it was living in the Substrate repo back then. Rialto then has been "upgraded" and forked from one of Polkadot chains (Rococo iirc). Rialto-parachain ahs been forked from one of parachains living in the Cumulus repo

[the remark this answers is not in the document]
Comment (svyatonik) [reply bin/millau/runtime/src/lib.rs:794]: > Or is it copied and adapted ?

Ideally - just copied. In practice - we're copying and then fixing compilation errors

[the remark this answers is not in the document]
Comment (serban300) [reply bin/millau/runtime/src/lib.rs:795]: Ok, thanks ! I'll think some more about this after the PR is merged. There is also a somewhat related issue: #1233. Resolving the comment for the moment.

**Review by serban300 — APPROVED** [review]

**Pull request merged**